scipy/linalg/decomp.py

```diff
@@ -1267,3 +1268,138 @@ def hessenberg(a, calc_q=False, overwrite_a=False, check_finite=True):
     q, info = orghr(a=hq, tau=tau, lo=lo, hi=hi, lwork=lwork, overwrite_a=1)
     _check_info(info, 'orghr (hessenberg)', positive=False)
     return h, q
+
+
+def cdf2rdf(w, v):
+    """
+    Converts complex eigenvalues ``w`` and eigenvectors ``v`` to real
+    eigenvalues in a block diagonal form ``wr`` and the associated real
+    eigenvectors ``vr``, such that::
+
+        vr @ wr = X @ vr
+
+    continues to hold, where ``X`` is the original array for which ``w`` and
+    ``v`` are the eigenvalues and eigenvectors.
+
+    Parameters
+    ----------
+    w : array_like
+        Complex or real eigenvalues array of size (M,) for a single matrix, or
+        size (..., M) for multiple stacked matrices
+    v : array_like
+        Complex or real eigenvectors array of size (M, M) for a single matrix,
+        or size (..., M, M) for multiple stacked matrices
+
+    Returns
+    -------
+    wr : ndarray
+        Real diagonal block form of eigenvalues, with size (M, M) for single
+        matrix, or (..., M, M) for multiple stacked matrices.
+    vr : ndarray
+        Real eigenvectors associated with ``wr``, with size (M, M) for
+        single matrix, or (..., M, M) for multiple stacked arrays.
+
+    Notes
+    -----
+    ``w``, ``v`` must be the eigenstructure for some *real* matrix ``X``.
+    For example, obtained by ``w, v = scipy.linalg.eig(X)`` or
+    ``w, v = numpy.linalg.eig(X)`` in which case ``X`` can also represent
+    stacked arrays.
+
+    Examples
+    --------
+    >>> X = numpy.array([[1, 2, 3], [0, 4, 5], [0, -5, 4]])
```

Comment:
use `np` instead of `numpy` in examples (and no need to import it). That the code itself in this file uses `numpy` is a historical artifact, and doesn't matter too much.